wordpress-mobile/WordPress-iOS
# Notifications: add mark all read button to nav bar

Ref: #17135 

With this PR, the following 2 steps are addressed. PR's will follow for the remaining 4:
- [x] Add button to nav bar.
- [ ] Mark notifications read for the selected filter.
- [ ] Add tracks for feature usage.
- [ ] Accessibility support.
- [ ] Enable feature flag and add release note (after branch is merged to develop).

To Test:
1. Enable the Feature Flag: Settings -> AppSettings -> Debug -> Enable "Mark Notifications As Read"
2. Notifications Tab.
3. A Checkmark button must have been placed left of the Settings Icon. The button will be non-functional until the follow up PR is merged.
4. Reader Tab's Bar Buttons' positions are also adjusted to match the Design Guidelines. They all have 44x44 tappable area now and should match in positions between the two tabs (Reader-Notifications).

## Regression Notes
1. Potential unintended areas of impact
The "Quick Start" feature displays an indicator on top of the button. I do not know how to trigger this case and I must test it. Some guidance/steps here would be very much appreciated.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
TODO, see step 1.

3. What automated tests I added (or what prevented me from doing so)
I couldn't find a UI Test file for `NotificationsViewController.swift`. Please let me know if I should write one.

PR submission checklist:


## Screen Shot for Reference:
<img width="470" alt="Screenshot 2022-01-15 at 16 44 01" src="https://user-images.githubusercontent.com/15968946/149627911-b03665f4-bec1-4195-adc8-7eca678e00b7.png">
